### PR TITLE
Update Vertoz adapter for Prebid 1.0

### PR DESCRIPTION
**Pull request by mohit546**
* Migrating to Prebid 1.0
        * spec added

## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'vertoz',
  params: {
    placementId: "VZ-HB-B784382V6C6G3C"
  }
}
```

### CODE REVIEW

File: modules/vertozBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'vertoz';
+const BASE_URI = '//hb.vrtzads.com/vzhbidder/bid?';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [], // short code
```
Comment (matthewlane) [inline modules/vertozBidAdapter.js]: If you don't have any aliases defined, this line can be dropped

File: modules/vertozBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        dealId: null,
+        creativeId: null,
+        ttl: 300,
+        ad: bidRespObj.ad + utils.createTrackPixelHtml(decodeURIComponent(bidRespObj.nurl))
+      };
+      bidResponses.push(bidResponse);
+    }
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions, serverResponses) {
```
Comment (matthewlane) [inline modules/vertozBidAdapter.js]: If you don't need any user syncing, this function can be removed

File: test/spec/modules/vertozBidAdapter_spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    ];
+
+    it('sends bid request to ENDPOINT via POST', () => {
+      const request = spec.buildRequests(bidRequests)[0];
+      expect(request.url).to.equal(BASE_URI);
+      expect(request.method).to.equal('POST');
+    });
+  });
+});
```
Comment (matthewlane) [inline test/spec/modules/vertozBidAdapter_spec.js]: These tests are only providing 68.97% coverage of the adapter, we require at least 80% before merge. Removing the unneeded code mentioned in the other comments should bump the percentage up, and then testing the `interpretResponse` should help getting there

**Comment by matthewlane** [comment]
@mohit546 Please also submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html)

**Comment by matthewlane** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/616

**Comment by Prebid-Vertoz** [comment]
Hi @matthewlane,
Can you please share request here so that we can find more.
Because we are unable to create this issue on our side.

**Comment by matthewlane** [comment]
@Prebid-Vertoz Sure. After running `gulp serve`, and placing the vertoz params on the [hello_world](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html#L19) test page, the request I'm seeing to http://hb.vrtzads.com/vzhbidder/bid? is

```
{"q":"%7B%22_vzPlacementId%22%3A%22VZ-HB-B784382V6C6G3C%22%2C%22_rqsrc%22%3A%22http%3A%2F%2Flocalhost%3A9999%2FintegrationExamples%2Fgpt%2Fhello_world.html%22%2C%22_cb%22%3A1518816756%2C%22_slotBidId%22%3A%2223d814c7aad797%22%2C%22_cbn%22%3A%22pbjs%22%7D"}
```

and the response is
```
{"slotBidId":"23d814c7aad797","statusText":"Application_Error"}
```

**Comment by Prebid-Vertoz** [comment]
Hi @matthewlane,
Cause behind getting "Application Error" in response is  "localhost:9999" in "_rqsrc" parameter. 
This issue is arising because we are using a third party library for domain validation and it is not allowing 
 ":" in domain. 
Is it possible to test it with some valid domain to get a proper response?

**Comment by matthewlane** [comment]
@Prebid-Vertoz Moved my test page to a domain without a port in the domain, request is now
```
{"q":"%7B%22_vzPlacementId%22%3A%22VZ-HB-B784382V6C6G3C%22%2C%22_rqsrc%22%3A%22http%3A%2F%2Fmlane.devnxs.net%2Fvertoz%2F%22%2C%22_cb%22%3A1519153843%2C%22_slotBidId%22%3A%224f809a85a752f3%22%2C%22_cbn%22%3A%22pbjs%22%7D"}
```

and response is now
```
{"slotBidId":"4f809a85a752f3","statusText":"InvalidRequest"}
```

Please advise about this error. If you have a publicly available test page that requests and returns bids with this adapter I can check that as well

**Comment by Prebid-Vertoz** [comment]
@matthewlane ,

**Comment by matthewlane** [comment]
@Prebid-Vertoz Great, I'm able to access that page. Current response from Vertoz there is

```
{"vzhPlacementId":"VZ-HB-B784382V6C6G3C","slotBidId":"21613f3032bf11","statusText":"NO_BIDS"}
```

please check. As soon as I can verify a valid bid response rather than a `no_bids` response this should be good for merge

**Comment by Prebid-Vertoz** [comment]
Hi @matthewlane , 
Please check now. You should get a bid in return.

**Comment by mohit546** [comment]
@matthewlane ,
tried to download vertoz adaptor 1.4.0 from http://prebid.org/
got email
selection_008

but getting following error on link
selection_009

**Comment by matthewlane** [comment]
@mohit546 Thanks for letting me know, the docs pr was merged before the download page got updated for 1.5.0. I reverted the commit and will add it back once the page is updated and it should work properly then

**Comment by mohit546** [comment]
@matthewlane i am not seeing vertoz adaptor in 1.12.0, and there is no option for 1.0.0. do i have to update adaptor to latest version ?

**Pull request merged**